spanner/value.go

```diff
@@ -3564,6 +3614,24 @@ func encodeValue(v interface{}) (*proto3.Value, *sppb.Type, error) {
 		pt = proto.Clone(v.Type).(*sppb.Type)
 	case []GenericColumnValue:
 		return nil, nil, errEncoderUnsupportedType(v)
+	case protoreflect.Enum:
+		var protoEnumfqn string
+		if v != nil {
+			pb.Kind = stringKind(strconv.FormatInt(int64(v.Number()), 10))
+			protoEnumfqn = string(v.Descriptor().FullName())
+		}
+		pt = enumType(protoEnumfqn)
```

Comment:
If v is nil, it seems protoEnumfqn is not set. Is this expected?

Do we have a test for this? If so, which line. Thanks!

Reply:
The only way we can pass nil here is to directly pass `nil` while inserting data. We cannot create a nil instance of an `enum` (`pb.Genre_Rock` or `pb.Genre`) or `proto` (`pb.Book{}`) as they create default objects in every possibility of declaration. We cannot get descriptor fn for `nil`. So assigning `""` in that case.

Reply:
Also this line will not be called as there is no way to send a nil instance of an enum or proto. I have followed the coding convention of client library and also added unit tests for the same.

Reply:
The only way to pass a nil type of proto enum is by type casting and passing like `(*pb.Genre)(nil)`. As it is a nil type, calling `v.Descriptor()` raises a panic in golang with exception as follows:

`panic: value method Genre.Descriptor called using nil *Genre pointer`

So, the only option is to set `protoEnumfqn` as `""`.
I have added a unit test for the same [Test](https://github.com/googleapis/google-cloud-go/pull/6886/files#diff-d66cf51d595f7a3cfd7349436181f0575be539d660acefae20b4d22fcfb96a6fR480)

Reply:
The following code will give the same type `(*pb.Genre)(nil)` as pointed above
```
v := pb.Genre_ROCK
vp := &v
vp = nil
```
golang throws an exception when any method is called using nil pointer. The only reasonable option would be to throw an error to user indicating that nil types cannot be used.
I have added the code changes @Shua1. Can you please have a look?